viewer/packages/tools/src/Timeline/TimelineTool.test.ts

```diff
@@ -0,0 +1,168 @@
+    TWEEN.update(current + 30000);
+
+    expect(assignStyledNodeCollectionSpy).toBeCalledWith(kf3Collection, kf3Appearance);
+    TWEEN.update();
+
+    expect(unassignStyledNodeCollectionSpy).toBeCalledWith(kf2Collection);
+  });
+
+  test('stop() stops the Timeline playback', () => {
+    const assignStyledNodeCollectionSpy = jest.spyOn(model, 'assignStyledNodeCollection');
+
+    timelineTool.play(new Date('2021-10-25'), new Date('2021-10-27'), 40000);
+    const current = TWEEN.now();
+    TWEEN.update(current + 10000);
+
+    expect(assignStyledNodeCollectionSpy).toBeCalledWith(kf1Collection, kf1Appearance);
+    TWEEN.update(current + 20000);
+
+    expect(assignStyledNodeCollectionSpy).toBeCalledWith(kf2Collection, kf2Appearance);
+    TWEEN.update(current + 30000);
+
+    timelineTool.stop();
+
+    expect(TWEEN.update()).toBeFalse();
+  });
+
+  test('pause() & resume() pauses & resumes the Timeline', () => {
+    timelineTool.play(new Date('2021-10-25'), new Date('2021-10-27'), 40000);
+    const current = TWEEN.now();
+    TWEEN.update(current + 10000);
+
+    expect(TWEEN.update()).toBeTrue();
+
+    timelineTool.pause();
+    expect(TWEEN.update()).toBeFalse();
+    TWEEN.update(current + 20000);
+
+    timelineTool.resume();
+    expect(TWEEN.update()).toBeTrue();
+    TWEEN.update(current + 30000);
+  });
+
+  test('play() while play is active', () => {
+    const stopSpy = jest.spyOn(timelineTool, 'stop');
+    const assignStyledNodeCollectionSpy = jest.spyOn(model, 'assignStyledNodeCollection');
+    const unassignStyledNodeCollectionSpy = jest.spyOn(model, 'unassignStyledNodeCollection');
+
+    timelineTool.play(new Date('2021-10-25'), new Date('2021-10-27'), 30000);
+    const current = TWEEN.now();
+    TWEEN.update(current + 10000);
+
+    timelineTool.play(new Date('2021-10-25'), new Date('2021-10-27'), 30000);
+
+    expect(stopSpy).toBeCalledTimes(2);
+
+    TWEEN.update(current + 20000);
+
+    expect(assignStyledNodeCollectionSpy).toBeCalledWith(kf1Collection, kf1Appearance);
+    TWEEN.update(current + 30000);
+
+    expect(assignStyledNodeCollectionSpy).toBeCalledWith(kf2Collection, kf2Appearance);
+    TWEEN.update(current + 40000);
+
+    expect(assignStyledNodeCollectionSpy).toBeCalledWith(kf3Collection, kf3Appearance);
+    TWEEN.update();
+
+    expect(unassignStyledNodeCollectionSpy).toBeCalledWith(kf2Collection);
+  });
+
+  test('stop() while in stopped', () => {
+    const assignStyledNodeCollectionSpy = jest.spyOn(model, 'assignStyledNodeCollection');
+
+    timelineTool.play(new Date('2021-10-25'), new Date('2021-10-27'), 40000);
+    const current = TWEEN.now();
+    TWEEN.update(current + 10000);
+
+    expect(assignStyledNodeCollectionSpy).toBeCalledWith(kf1Collection, kf1Appearance);
+    TWEEN.update(current + 20000);
+
+    expect(assignStyledNodeCollectionSpy).toBeCalledWith(kf2Collection, kf2Appearance);
+    TWEEN.update(current + 30000);
```

Comment:
This is not part of what you really are testing in this test and should be removed.